From dfe4e3f48df49dda644d05a47acf35e9e7e59491 Mon Sep 17 00:00:00 2001
From: Nix Automated Build <noreply@example.com>
Date: Tue, 28 Jul 2020 18:22:35 +0200
Subject: [PATCH] Fix anybar

---
 dwm.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/dwm.c b/dwm.c
index d1a9e08..f4e9b34 100644
--- a/dwm.c
+++ b/dwm.c
@@ -699,9 +699,6 @@ configurerequest(XEvent *e)
 		wc.width = ev->width;
 		wc.height = ev->height;
 		wc.border_width = ev->border_width;
-	if (usealtbar)
-		return;
-
 		wc.sibling = ev->above;
 		wc.stack_mode = ev->detail;
 		XConfigureWindow(dpy, ev->window, ev->value_mask, &wc);
@@ -801,6 +798,9 @@ dirtomon(int dir)
 void
 drawbar(Monitor *m)
 {
+	if (usealtbar)
+		return;
+
 	int x, w, sw = 0;
 	int boxs = drw->fonts->h / 9;
 	int boxw = drw->fonts->h / 6 + 2;
-- 
2.27.0

